### PR TITLE
Move gMapSize to GameState_t

### PR DESCRIPTION
**Pull request by Harry-Hopkinson**
For https://github.com/OpenRCT2/OpenRCT2/issues/21193

### CODE REVIEW

File: src/openrct2-ui/windows/Map.cpp [hunk 1]
```diff
@@ -162,7 +163,8 @@ class MapWindow final : public Window
         CentreMapOnViewPoint();
         FootpathSelectDefault();
 
-        _mapWidthAndHeightLinked = gMapSize.x == gMapSize.y;
+        auto& gameState = OpenRCT2::GetGameState();
```
Comment (Harry-Hopkinson) [inline src/openrct2-ui/windows/Map.cpp:166]: Should I change to use a namespace as OpenRCT2 is referenced a lot in this file?

File: src/openrct2/world/Map.cpp [hunk 2]
```diff
@@ -129,11 +127,27 @@ void UnstashMap()
 {
     _tileIndex = std::move(_tileIndexStash);
     _tileElements = std::move(_tileElementsStash);
-    gMapSize = _mapSizeStash;
+    GetGameState().MapSize = _mapSizeStash;
     gCurrentRotation = _currentRotationStash;
     _tileElementsInUse = _tileElementsInUseStash;
 }
 
+CoordsXY GetMapSizeUnits()
```
Comment (Harry-Hopkinson) [inline src/openrct2/world/Map.cpp:135]: Had to move from an inline function as it references the static gameState variable.

Comment (ZehMatt) [reply src/openrct2/world/Map.cpp:135]: That's fine, not sure why it was done in the header anyway.

File: src/openrct2-ui/windows/Map.cpp (hunk 1 above)
Comment (ZehMatt) [reply src/openrct2-ui/windows/Map.cpp:166]: Yeah give it a try, there might be a problem but when it doesn't cause any compiler errors then its actually good as we had planned to move everything into the OpenRCT2 namespace eventually.

**Review by Gymnasiast — APPROVED** [review]

**Pull request merged**